docs/reference/how-to/search-speed.asciidoc

```diff
@@ -500,6 +500,36 @@ enough to hold all the data. Use with caution.
 conjunctions faster at the cost of slightly slower indexing. Read more about it
 in the <<index-modules-index-sorting-conjunctions,index sorting documentation>>.
 
+[float]
+=== Use routing to trade latency for throughput
+
+By default, documents get randomly assigned to a shard. This has the side-effect
+of giving all shards similar content and in particular, a similar dictionary of
+terms. Now imagine that you are indexing a catalog of products where each
+product belongs to exactly one category, and that you
+<<mapping-routing-field,route>> documents using this `category` field. Each
+category will exist on exactly one shard. The side-effect is that many searches
+will have most of their hits on one shard while other shards will have very few
+hits, possibly even no hits at all. For instance a query on `pizza` likely only
+has hits in the `Food` category, so the shard that has all documents from the
+`Food` category will have all hits while other shards will return an empty
+response instantly since `pizza` doesn't exist in their terms dictionary.
+
+Given that the latency of a search is the latency of the slowest shard, this
```

Comment:
This seems like a helpful addition!

I wonder if we could give users clearer guidance on when using routing could help. As it currently reads, I think it'd be hard for a user to know whether it's worth testing out a custom routing strategy (especially given the downsides). You mention some of this implicitly, but maybe we could explicitly call out some cases like the following:
* There's a large number of unique terms and your searches make use of `fuzzy` or `regex` queries.
* The data is structured in a way such that you can use search routing to avoid fanning out to all shards.